src/sentry/migrations/0028_discover_query_upgrade.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+# Generated by Hand
+from __future__ import unicode_literals, print_function
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+from sentry.discover.models import DiscoverSavedQuery, DiscoverSavedQueryProject
+from sentry.api.event_search import SearchVisitor
+from django.db import migrations
+
+# Aggregates are now fields
+def convert_field(fieldname, unique, reverse):
+    if fieldname == "count":
+        fieldname = u"count()"
+    elif unique:
+        fieldname = u"count_unique({})".format(fieldname) if unique else field
+
+    fieldname = u"-{}".format(fieldname) if reverse else fieldname
+    return fieldname
+
+def prepare_value(value):
+    value = value.replace("%", "*")
+    if " " in value and not value.startswith('"'):
+        value = u'"{}"'.format(value)
+    return value
+
+def convert(saved_query, name_extra=" (migrated from legacy discover)"):
+    """ Create a v2 query from a v1 query"""
+    if saved_query.version == 2:
+        # nothing to do! Already v2 :)
+        return saved_query
+
+    updated_query = {
+        u"environment": [],
+        u"fields": saved_query.query.get('fields', []),
+        u"orderby": u"",
+        u"query": [],  # Will become a string later via join
+    }
+
+    if "range" in saved_query.query:
+        updated_query["range"] = saved_query.query["range"]
+    elif "start" in saved_query.query and "end" in saved_query.query:
```

Comment:
If there is a start is there always guaranteed to be an end? You can't have a start and a range?

Reply:
Based on the data, the 68 rows with starts always have an `end` :+1:
There's 5 with `start` and a `range` but range is an empty string in those cases

Reply:
I expect start, end, and range to be converted in the same precedence order as this function https://github.com/getsentry/sentry/blob/d199c9a2dda5bdf85627291d216d5caecd960222/src/sentry/api/utils.py#L33-L84

Reply:
TBH I could probably just forcibly use that function, but there doesn't exist any queries that require using it. And I think it makes more sense if we use range over start/end since start/ends stop working once they go beyond retention.

Reply:
Ah. I didn't mean to suggest that this function be used. 😅 

I think the implementation you had was fine. 👍